### PR TITLE
[migration] Add clarification for validate() for header only

### PR DESCRIPTION
**Pull request by danimtb**
closes https://github.com/conan-io/conan/issues/12210

### CODE REVIEW

**Review by memsharded — APPROVED** [review]

File: migrating_to_2.0/recipes.rst [hunk 1]
```diff
@@ -205,6 +205,16 @@ are valid.
...
                 raise ConanInvalidConfiguration("This package is not compatible with Windows")
 
+.. note::
+
+    For recipes where settings are cleared, using ``self.settings`` is still valid. For example,
+    this applies to header only recipes that check for a specific ``self.settings.cppstd`` like:
+
+    .. code-block:: python
+
+        def validate(self):
```
Comment (uilianries) [inline migrating_to_2.0/recipes.rst:215]:
```suggestion
        def package_id(self):
            self.info.clear()

        def validate(self):
```
Make it more explicit according to the note. Anything removing compiler from the package ID should follow the same path of course.

**Comment by prince-chrismc** [comment]
🤔  Trying to see why my PR is failing the 2.0 pipeline... I hit this issue 

https://github.com/conan-io/conan-center-index/blob/318053389c781d02566c2609e9e77a1e205ca00a/recipes/expected-lite/all/conanfile.py#L23-L25

Fails with 2.0 with `'self.settings' access in 'validate()' method is forbidden`

```
string-view-lite/1.6.0: Compatible package ID da39a3ee5e6b4b0d3255bfef95601890afd80709 equal to the default package ID
variant-lite/2.0.0: Compatible package ID da39a3ee5e6b4b0d3255bfef95601890afd80709 equal to the default package ID
ERROR: restinio/0.6.17: Error in validate() method, line 100
        check_min_vs(self, 190)
        ConanException: 'self.settings' access in 'validate()' method is forbidden
 ✘ christopherm@christopherm-mac  ~/conan-center-index/recipes/restinio   update/restinio ±  conan --version
Conan version 2.0.0-beta3
```

File: migrating_to_2.0/recipes.rst [hunk 2]
```diff
@@ -205,6 +205,19 @@ are valid.
...
+    For recipes where settings are cleared, using ``self.settings`` is still valid. For example,
+    this applies to header only recipes that check for a specific ``self.settings.cppstd`` like:
+
+    .. code-block:: python
+
+        def package_id(self):
+            self.info.clear()
+
+        def validate(self):
+            if self.settings.get_safe("compiler.cppstd"):
```
Comment (SSE4) [inline migrating_to_2.0/recipes.rst:219]: does it work correctly in 2.x?

Comment (uilianries) [reply migrating_to_2.0/recipes.rst:219]: Yes:

```
$ conan --version
Conan version 2.0.0-beta4
```

```python
from conan import ConanFile
from conan.tools.build import check_min_cppstd

class PackageConan(ConanFile):
    settings = "os", "arch", "compiler", "build_type"

    def validate(self):
        if self.info.settings.get_safe("compiler.cppstd"):
            check_min_cppstd(self, 17)

```

```
$ conan create . --name=foo --version=0.1
Exporting the recipe
foo/0.1: Exporting package recipe
foo/0.1 exports: File 'conandata.yml' found. Exporting it...
foo/0.1: Copied 1 '.yml' file: conandata.yml
foo/0.1 exports: Copied 1 '.yml' file: conandata.yml
foo/0.1: Using the exported files summary hash as the recipe revision: e1717e578ab2a323f66a39375f08ede5 
foo/0.1: A new conanfile.py version was exported
foo/0.1: Folder: /Users/uilian/.conan2/p/8ee45637f11fbb14/e
foo/0.1: Exported revision: e1717e578ab2a323f66a39375f08ede5

-------- Input profiles --------
Profile host:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=14
compiler.libcxx=libc++
compiler.version=13
os=Macos

Profile build:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=14
compiler.libcxx=libc++
compiler.version=13
os=Macos


-------- Computing dependency graph --------
Graph root
    virtual
Requirements
    foo/0.1#e1717e578ab2a323f66a39375f08ede5 - Cache

-------- Computing necessary packages --------
foo/0.1: Forced build from source
Requirements
    foo/0.1#e1717e578ab2a323f66a39375f08ede5:72bcad4ae058f9afaffb458359f4f4f45523a218 - Invalid

-------- Installing packages --------

-------- Installing (downloading, building) binaries... --------
ERROR: There are invalid packages (packages that cannot exist for this configuration):
foo/0.1: Invalid: Current cppstd (14) is lower than the required C++ standard (17).
```

**Review by SSE4 — APPROVED** [review]

**Comment by prince-chrismc** [comment]
This needs some attention https://github.com/conan-io/docs/pull/2778/files#diff-633dc3f2388dc1689eaab8f6f51e1435bf9be9a28e73d49d302a583de6360789R185 should be more clear about `validate` vs `package_id`

**Comment by prince-chrismc** [comment]
This PR matches the current recommendations and it's doesn’t have any open questions AFAIK so I will go ahead and merge this tiny snippet

**Pull request merged**